lib/src/realm_object.dart

```diff
@@ -411,7 +411,17 @@ mixin RealmObjectBase on RealmEntity implements Finalizable {
 mixin RealmObject on RealmObjectBase {}
 
 /// @nodoc
-mixin EmbeddedObject on RealmObjectBase {}
+mixin EmbeddedObject on RealmObjectBase {
+  RealmObjectBase? getParent() {
```

Comment:
Should we make this a `parent` getter instead?

Reply:
+1

Reply:
I don't think it's a good idea, because users may want to use `parent` as a property on their models.

Reply:
Had a long discussion about this with @nirinchev and we settled on trying with a getter